app/assets/javascripts/sitewide/atwho_res.js.erb

```diff
@@ -136,16 +148,31 @@ var SmartAnnotation = (function() {
 
     // Generates suggestion dropdown filter
     function generateFilterMenu(active, res_data) {
+      var rep_buttons = '';
+      $.ajax({
+        async: false,
+        dataType: 'json',
+        url: $(document.body).attr('data-atwho-repositories-url'),
+        success: function(data) {
+          $.each(data['repositories'], function(id, name) {
+            if(defaultRepId === undefined){
+              defaultRepId = id;
+            }
+            rep_buttons += '<button data-filter="rep" data-rep-id="' + id +
+              '" class="btn btn-xs btn-primary">' + name + '</button>';
+          });
+       }
+      });
       var header = '<div class="atwho-header-res">' +
-       '<div class="title">' +
-       '<button data-filter="prj" class="btn btn-xs ' +
-       (active === 'prj' ? 'btn-primary' : 'btn-default') + '">project#</button>' +
-       '<button data-filter="exp" class="btn btn-xs ' +
-       (active === 'exp' ? 'btn-primary' : 'btn-default') + '">experiment#</button>' +
-       '<button data-filter="tsk" class="btn btn-xs ' +
-       (active === 'tsk' ? 'btn-primary' : 'btn-default') + '">task#</button>' +
-       '<button data-filter="sam" class="btn btn-xs ' +
-       (active === 'sam' ? 'btn-primary' : 'btn-default') + '">sample#</button>' +
+       '<button data-filter="prj" class="btn btn-xs btn-primary' +
```

Comment:
If you want you can extract the HTML in the separate JS module that is purely functional and returns HTML snippets. is nothing wrong to generate HTML with JS but we can separate the concerns and clean up